src/prepared.jl

```diff
@@ -0,0 +1,56 @@
+struct Prepared{Pa,Pr}
+    parent::Pa
+    preparations::Pr
+end
+
+Base.parent(x::Prepared) = x.parent
+@inline getprep(p::Prepared, x::Symbol) = getpropery(p.preparations, x)
+
+GI.trait(p::Prepared) = GI.trait(parent(p))
+GI.geomtrait(p::Prepared) = GI.geomtrait(parent(p))
+
+GI.isgeometry(::Type{<:Prepared{T}}) where {T} = GI.isgeometry(T)
+GI.isfeature(::Type{<:Prepared{T}}) where {T} = GI.isfeature(T)
+GI.isfeaturecollection(::Type{<:Prepared{T}}) where {T} = GI.isfeaturecollection(T)
+
+GI.geometry(x::Prepared) = GI.geometry(parent(x))
+GI.properties(x::Prepared) = GI.properties(parent(x)) 
+
+for f in (:extent, :crs)
+    @eval GI.$f(t::GI.AbstractTrait, x::Prepared) = GI.$f(t, parent(x))
+end
+for f in (:coordnames, :is3d, :ismeasured, :isempty, :coordinates, :getgeom)
+    @eval GI.$f(t::GI.AbstractGeometryTrait, geom::Prepared, args...) = GI.$f(t, geom, args...)
```

Comment:
Should this be 
```suggestion
    @eval GI.$f(t::GI.AbstractGeometryTrait, geom::Prepared, args...) = GI.$f(t, parent(geom), args...)
```
?  Or are you going for something else here